Redesigned Installer box + Rounded UI + Other UI Improvements

UltimatChamp opened the pull request:
## [CHECK NEWER COMMENTS FOR UPDATES]

The stable 1.20.1 installer-dropdown is open by default, while the beta version will be closed.
The badges will also guide the user to install the correct version.

**Here are some images:**


_DESKTOP_
1) Light
<img src="https://i.imgur.com/muJeMI8.png">

2) Dark
<img src="https://i.imgur.com/iB8ACj7.png">


_MOBILE_
1) Light
<img src="https://i.imgur.com/NEBmDG2.png">

2) Dark
<img src="https://i.imgur.com/rIpvKwb.png">

<hr>

 Also when hovering the dropdowns: 
1) Light
<img src="https://i.imgur.com/iODLxGJ.png">

2) Dark
<img src="https://i.imgur.com/tNUMGlx.png">

------------------
Preview URL: https://pr-15.neoforged-website-previews.pages.dev

UltimatChamp commented:
The color of the recommended badge may need to change. Any suggestions. 

Also, the text color of the badges is made different, for visibility, in dark mode for recommend.

I may also set the text color to white in both the light and dark modes.

UltimatChamp commented:
I will also make the badges more elegant by separating them from the version-name in summary tag. (by not centering the badges too)

sciwhiz12 commented on assets/js/neoforge.js:
```diff
@@ -9,14 +9,26 @@ async function loadLatestVersions(minecraftVersions) {
+		mcvers = "1.20.1";
+        note = `<b>Note:</b> This file is still called <i>forge</i> because we're trying to maintain compatibility with launchers, assuming they don't hardcode things too much.`;
+        dropDown_VAL = ` open="open"`;
+        badges_one = `<font class="badges_stable">RECOMMENDED</font>`;
```
I don't know of a good reason why 1.20.1 would be marked as "recommended". We don't recommend 1.20.1 over any other version.

UltimatChamp replied:
Because 1.20.2 is still in beta. OK, I will change recommend with Stable.

sciwhiz12 replied:
I would prefer not having a badge for 1.20.1 that remarks it as either recommended or stable. There's nothing particularly important about 1.20.1 which merits a badge.

On the other hand, NF 20.1 merits a `BETA` badge because it is in beta.

UltimatChamp commented:
Instead of hard-coding the 20.2 releases as beta, I'm going to make it dynamic, so that the website do not need to change when stable versions start releasing.

UltimatChamp commented:
Here's the final pull:
<img src="https://i.imgur.com/HL8T3Tx.png">
<img src="https://i.imgur.com/XiEfIhg.png">

UltimatChamp commented:
Here are the changes: 
Home Page:
![DagFZrd](https://github.com/neoforged/websites/assets/140878474/0f87338e-6c7d-4f65-a154-accde94dfa40)
All Versions page:
![BDbX5mr](https://github.com/neoforged/websites/assets/140878474/6913d4e5-0bd9-4399-9dfc-3f09f77380c0)

UltimatChamp commented:
Fixes Issue: https://github.com/neoforged/websites/issues/25

Technici4n commented:
Well I didn't see this. Fixed in a different way.

UltimatChamp commented:
The <a href="https://projects.neoforged.net/neoforged/neoforge">projects.neoforged.net</a> page is very helpful. Not drafting the pull, since, the redesigned installer box is still a good addition. 
Changing the pull to integrate this new page.

UltimatChamp commented:
Integrated the <a href="https://projects.neoforged.net/neoforged/neoforge">projects.neoforged.net</a> page and also, redesigned the `All Versions` button.
![image](https://github.com/neoforged/websites/assets/140878474/e65e1500-cfb4-4f44-9b9a-0b3bcf7add62)

Technici4n commented on content/_index.md:
```diff
@@ -12,11 +12,10 @@ description: |
-{{<projects>}}
-
+<!--Change with the latest version here-->
+{{< files "20.4" >}}
```
Do we want to hardcode this latest version here or could we maybe fetch it from the maven somehow?

UltimatChamp replied:
Will be very helpful; like the `projects` website. Will do some research...

Technici4n commented on layouts/shortcodes/versions.html:
```diff
@@ -0,0 +1,3 @@
+<div class="versions-border">
```
Do you think it would be nice to make this look a bit more like the download buttons? (Like I did for the current version of the webpage)

UltimatChamp replied:
Don't think it will look good now due to the `details` tag.

UltimatChamp replied:
I tried to fit it in, by making it look like the search field.

UltimatChamp replied:
Meeting after 2 weeks since I've been really busy. As requested, the changes have been made in https://github.com/neoforged/websites/pull/15/commits/04be4272934440318ca5847a1c39477a4d8de1dd.

Technici4n commented on assets/js/neoforge.js:
```diff
@@ -9,14 +9,24 @@ async function loadLatestVersions(minecraftVersions) {
+        fn = "neoforge";
+		mcvers = `1.${mcVersion}`;
+        badges_beta = "";
+        badges_new = `<font class="badges_new">NEW</font>`;
```
We probably only want this for the latest release? (assuming we fetch that info automatically, see other comment)

UltimatChamp replied:
Ah, yes! Now it's only for a single version; the latest one.

Technici4n reviewed (commented):
I like that design and I think the badges look really cool, just a few questions. :)

UltimatChamp commented on layouts/shortcodes/files.html:
```diff
@@ -4,13 +4,17 @@
   {{ .Page.Scratch.Set "nfJS" "hi" }}
 {{ end }}
 
-<div id="filelist{{ replace (.Get 0) "." "" }}" class="fileinfo">
```
Well, seems to be a Github issue because here on VS Code it seems to be OK.

UltimatChamp commented:
Found an easy way to fetch the latest version! Now the website doesn't need to update when new versions come out.

UltimatChamp commented:
Everything in 0be0287eb7cb18f8e0a0b7fea0c9201565fdfd28:

1) Fixed an issue where the notice boxes follow only the system's/browser's color-scheme/theme, instead of the website's theme (when the theme is toggled in the website); now layout 'box' will be used instead of 'notice' to call a tip/warning/note/...; this means that the 'hugo-notice' submodule can be used, without the need of just copying the whole repo into the website just to bring some small modifications.

2) Fixed the Mastodon icon to follow other social icon's color scheme.

3) Some more rounded UI implementations.

4) Updated the 'hugo-notice' submodule to the newest commit.

5) Modified the the warning notice to not look odd, from other notices.

Technici4n reviewed (changes requested):
Could you maybe change the mouse to a hand when hovering the text in the dropdowns? Might also be worth making the text non-selectable.

UltimatChamp commented:
> Could you maybe change the mouse to a hand when hovering the text in the dropdowns? Might also be worth making the text non-selectable.

Done!

Technici4n commented on layouts/shortcodes/box.html:
```diff
@@ -0,0 +1,17 @@
+<!--Notice Color Scheme Fix-->
+<!--Credits: https://github.com/martignoni/hugo-notice-->
```
The license is GPLv3, this might be problematic to use?

UltimatChamp replied:
Hmm... 
> Everyone is permitted to copy and distribute verbatim copies of this license document, but changing it is not allowed.

Shouldn't be a problem for such a small modification... Moreover, the repo has been properly credited as a submodule.

Technici4n replied:
The problem is that we would have to make all the code of the website GPL to comply with the license terms. I don't know if that is a problem or not.

UltimatChamp replied:
Can't we just mark the file as GPLv3 instead (like just mentioning in it)... If not, then I could just remove the file for the time being.

Technici4n replied:
The mainroad theme is actually GPL2, so probably OK? I'll think more about this and come back to you.

Technici4n replied:
Well we are already using a hugo-notice submodule so whatever.

Technici4n commented on assets/js/neoforge.js:
```diff
@@ -3,23 +3,39 @@ const FORGE_GAV = 'net/neoforged/neoforge'
 	let gav;
 	let fn;
 	let mcvers;
+        let dropDown_VAL;
```
Might want to change this for consistent indentation (you have mixed tabs and spaces, you can see it if you set github tabs to 4). But I don't mind too much, happy to merge without this change.

UltimatChamp replied:
Sighhh... This has been really annoying! VS Code and Github... Can you help me check again...? (pushing a commit)
